components/provisioner/assets/database/provisioner.sql

```diff
@@ -12,6 +12,26 @@ CREATE TABLE cluster
     sub_account_id varchar(256)
 );
 
+-- OIDC config
+
+CREATE TABLE oidc_config
+(
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    cluster_id uuid NOT NULL,
+    client_id text NOT NULL,
+    groups_claim text NOT NULL,
+    issuer_url text NOT NULL,
+    username_claim text NOT NULL,
+    username_prefix text NOT NULL
+);
+
+CREATE TABLE signing_algorithms
+(
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    cluster_id uuid NOT NULL,
+    algorithm text NOT NULL
+);
```

Comment:
There is no need to have `cluster_id` here as we keep the OIDC config ID in `gardener_config`. In the `signing_algorithms` should be the ID of the oidc_config as a foreign key as the algorithms are part of it. When the OIDC config is deleted so should be the signing_algorithms

I think it may cause little changes also in the implementation part

Reply:
Done. 
I added to `oidc_config` FK  gardener_config_id
and to `signing_algorithms` FK oidc_config_id